frontend/app/routes/_gcweb-app._index.tsx

```diff
@@ -36,23 +36,23 @@ export default function Index() {
         {t('common:index.page-title')}
       </h1>
       <p>
-        Welcome {userInfo.firstName} {userInfo.lastName}
+      {t('common:index.welcome')} {userInfo.firstName} {userInfo.lastName}
```

Comment:
Translations that include data should have the data inside of the translation.. like this:

`"my-string": "Welcome <firstName>John</firstName> <lastName>Doe</lastName>"`

and then the data should be substituted into the string using a `<Trans>` component.

The reason is that english and french have different grammar rules. The example I always use is this:

```
en: "The red <thing>object</thing>"
fr: "Le <thing>object</thing> route"
```

Reply:
@gregory-j-baker, only use `Trans` from `react-i18next` if you need to render html which is not the case here. The basic `i18next` interpolation should be used here.

https://www.i18next.com/translation-function/interpolation#basic

Keys, by default, are strings surrounded by curly brackets:
```json
"welcome": "Welcome {{firstName}} {{lastName}}",
```

`t` function call:
```ts
t('common:index.welcome', { firstName: userInfo.firstName, lastName: userInfo.firstName });
```

Trans Component
https://react.i18next.com/latest/trans-component

> **Important note**
> While <Trans> gives you a lot of power by letting you interpolate or translate complex React elements, the truth is: in most cases you don't even need it.

Reply:
Ahh yes. I actually completely forgot that you could do this with the `t(..)` function. Thanks for the tips.